r/src/dataset.cpp

```diff
@@ -58,12 +61,28 @@ std::shared_ptr<ds::SourceFactory> dataset___FSSFactory__Make3(
 }
 
 // [[arrow::export]]
-std::shared_ptr<ds::ParquetFileFormat> dataset___ParquetFileFormat__Make() {
-  return std::make_shared<ds::ParquetFileFormat>();
+std::shared_ptr<ds::ParquetFileFormat> dataset___ParquetFileFormat__Make(List_ options) {
+  auto fmt = ds::ParquetFileFormat::Make();
+  if (options.containsElementNamed("reader_options")) {
+    List_ reader_options = options["reader_options"];
+    if (reader_options.containsElementNamed("use_buffered_stream")) {
+      fmt->reader_options.use_buffered_stream = reader_options["use_buffered_stream"];
+    }
+    if (reader_options.containsElementNamed("buffer_size")) {
+      fmt->reader_options.buffer_size = reader_options["buffer_size"];
+    }
+    if (reader_options.containsElementNamed("dict_columns")) {
+      List_ dict_columns = reader_options["dict_columns"];
+      for (String name : dict_columns) {
+        fmt->reader_options.dict_columns.insert(name);
+      }
+    }
+  }
+  return fmt;
 }
 
 // [[arrow::export]]
-std::shared_ptr<ds::IpcFileFormat> dataset___IpcFileFormat__Make() {
+std::shared_ptr<ds::IpcFileFormat> dataset___IpcFileFormat__Make(List_ options) {
```

Comment:
There's probably no reason to pass the options in yet if we aren't doing anything with them

Reply:
Alright